add option to ensure directory for dest file exists

mojavelinux opened the pull request:
- mkdir:true ensures the directory exists using fs.mkdir with the recursive option
- mkdir:false does not check for the existence of the directory (default)

resolves #78

mcollina commented:
it seems the test is failing on windows.

mojavelinux commented:
Yep, I'm familiar with this kind of error. I'll address it.

mojavelinux commented:
If you approve the CI job to proceed, you should find that it is now fixed. (The CI job in my fork succeeded).

mcollina reviewed (approved):
lgtm, we don't need to add any new depedencies

mcollina commented:
This conflicts now, could you fix them?

jsumners commented on README.md:
```diff
@@ -58,6 +58,7 @@ The options are:
 * `fd`: a file descriptor, something that is returned by `fs.open` or
    `fs.openSync`.
 * `dest`: a string that is a path to a file to be written to (mode `'a'`).
+* `mkdir`: ensure directory for dest file exists.
```
```suggestion
* `mkdir`: ensure directory for dest file exists when `true`. Default: `false`.
```

mojavelinux replied:
Thanks for the suggestion. I'll apply it when I update the MR.

mojavelinux commented:
Yep, I was expecting the conflict. I'll update asap!

mojavelinux commented:
I have rebased. I also fixed a typo in the worth "length" in the README. If you prefer that change to be made in a separate PR, just let me know.

As a word of warning about the test, Windows has known timing issues when removing directories that once contained files. In short, fs sync is not really sync on Windows. This complicates the teardown routine. I could update the teardown logic to be more resilient against this problem if you'd like.

mcollina reviewed (approved):
lgtm, this seems ok

mojavelinux commented:
The build failed on Windows when this was merged into master due to the word of warning I gave in my last comment (Windows requires a more brute force approach to deleting a directory, which we have to do for the test). I'm happy to send a PR with the fix that I proposed. Just give me the :+1:

mcollina commented:
go for it!

mojavelinux commented:
On it! :runner:
